### ISSUE TITLE
AD - Enable with AD User, log out and log back in with AD credentials, but it shows local admim

### ISSUE THREAD

**Issue by deniseschannon** [rancher/rancher#12438]
Editted for the confusing use case based on logging in with a specific principal but showing the other principal. 

Enable AD with AD user "testuser1". After enabling, you will be logged in as test user 1.

Log out
Log in with "testuser1" credentials

Result: It will now look like you are logged in as the "default admin" for a local user. 

<img width="230" alt="screen shot 2018-04-02 at 8 16 44 pm" src="https://user-images.githubusercontent.com/11410997/38227383-ccfb9e5c-36b2-11e8-89b2-e2e48510ac2a.png">

Expected Results:

1. When logging in with "testuser1", you'd be logged in as "testuser1" instead of the "default admin".

**Comment by vincent99** [comment]
No; you're not creating any new users, you're adding a principal to the user you're already logged in as.  "testuser1" and "default admin" are the same user.

**Comment by deniseschannon** [comment]
If this is the case, then we need better explanation in the UI of what is happening when you enable an auth provider cause it's not clear. 

I'll make it a UI bug.

**Comment by vincent99** [comment]
Again, they are not "shared", they are the *same account*.  Everything is split up into 2 parts.

Resources inside of rancher are owned by Accounts.  Accounts have one or more principal IDs associated with them.  When you login with an auth provider it produces a principal ID (a github user number, a LDAP DN, a local username, etc) and an account having that principal ID associated to it is found.

When you enable a new form of access control and test it, the principal ID for your provider account you just tested is added to the account that is currently signed in.  There is _not_ a second account created.  This preserves your access to all the stuff you previously had access to.  Otherwise if you setup Rancher, you're admin, you go turn on Github, and then your github login would get you a new empty account with no access to anything.

The main thing that's missing here is the ability for other users (besides the one turning on auth) to see and change their associations to each auth provider that is enabled in the User preferences page.

**Comment by deniseschannon** [comment]
Shouldn't we be consistent and if you sign in with your principal ID, that's the one that shows up in the top menu or do we always plan to only show the account? 

I'd only want to fix this use case: 

```
When logging in with "testuser1", you'd be logged in as "testuser1" (prinicpal) instead of the "default admin" (user).
```

Is that possible?

**Comment by vincent99** [comment]
There's a single name on the Account, which is populated from the initial principal (or what someone types in for Local) on account creation.  Showing which principal you are logged in as for you is posisble, but the accounts table can't really show all possible principals for each account and we don't store the display names for any of the remote ones.  So anything that's not our local name field is going to show inconsistently to other people..

**Comment by deniseschannon** [comment]
Fixed in v2.0.3-rc2

**Comment by tulanian** [comment]
> Fixed in v2.0.3-rc2

What exactly was fixed? The behaviour is still the same in v2.2.7, and the UI does not say that this "linkage" will occur. Section 3 of Enabling AD Auth says only "Test and enable authentication," not "Link this account with the Admin account."

**Comment by deniseschannon** [comment]
What was fixed in the UI is that when you log in as the AD user, it will show the AD user details instead of the "Default Admin".

<img width="456" alt="Screen Shot 2019-08-21 at 11 20 02 AM" src="https://user-images.githubusercontent.com/11410997/63457340-b7db6480-c405-11e9-80b6-09caa808fc5a.png">

How the local default admin and the account you use to authenticate your external provider is linked together is explained in more detail in our docs:

https://rancher.com/docs/rancher/v2.x/en/admin-settings/authentication/#external-authentication-configuration-and-principal-users
